tests/ClientEndpointTest.php

```
@@ -0,0 +1,95 @@
+            'consumerSecret' => CONSUMER_SECRET,
+        ];
+        $this->log = Client::LOG_NONE;
+    }
+
+    /**
+     * Test create client with invalid URL
+     *
+     * @return void
+     */
+    public function testInvalidURLEndpoint(){
+
+        $this->baseConfig['apiEndpoint'] = '123abc';
+        try {
+            Client::create($this->baseConfig, $this->log);
+        } catch (\Exception $e) {
+            //throw $th;
+            $this->assertStringContainsString('Your API endpoint is not a valid URL', $e->getMessage());
+        }
+    }
+
+     /**
+     * Test create client with valid URL,
+     *
+     * @return void
+     */
+    public function testValidUrlWrongEndpoint(){
+
+        $this->baseConfig['apiEndpoint'] = 'https://api.carwash.com/v1/';
+
+        try {
+            $client = Client::create($this->baseConfig, $this->log);
+
+            $method = new Payment\Create($this->getPaymentParams());
+            $client->call($method);
+
+        } catch (\Exception $e) {
+            //throw $th;
+            $this->assertStringContainsString('error', $e->getMessage());
+        }
+        $this->assertInstanceOf('Cardinity\Client', $client);
+    }
+
+    /**
+     * Test create client with valid URL,
+     *
+     * @return void
+     */
+    public function testValidURLValidEndpoint(){
+
+        $this->baseConfig['apiEndpoint'] = 'https://api.cardinity.com/v1/';
+        try {
+            $client = Client::create($this->baseConfig, $this->log);
+        } catch (\Throwable $th) {
+            //throw $th;
+        }
+        $this->assertInstanceOf('Cardinity\Client', $client);
+    }
+
+     /**
+     * Test create client with valid URL,
+     *
+     * @return void
+     */
+    public function testNoEndpoint(){
+
+        unset($this->baseConfig['apiEndpoint']);
+        try {
+            $client = Client::create($this->baseConfig, $this->log);
+        } catch (\Throwable $th) {
+            //throw $th;
```

What is the reason for silently catching exception?